src/payment/models/buy-crypto/services/buy-crypto.service.ts

```diff
@@ -77,7 +77,7 @@ export class BuyCryptoService {
 
     // activate user
     if (entity.amlCheck === AmlCheck.PASS && entity.buy?.user?.status === UserStatus.NA) {
```

Comment:
Würde den Check `user?.status === UserStatus.NA` auch in das `activateUser()` verschieben.